pkg/api/types.go

```diff
@@ -2700,6 +2725,10 @@ type ServiceSpec struct {
 	// +optional
 	SessionAffinity ServiceAffinity
 
+	// ServiceAffinityConfig contains the configurations of session affinity.
+	// +optional
+	ServiceAffinityConfig *ServiceAffinityConfig
```

Comment:
Currently, I adopt the **embedded** way, and the API definition is

```
type ServiceSpec {
...
ClientIPAffinityConfig *ClientIPAffinityConfig
}
```

```
// ClientIPAffinityConfig represents the configurations of Client IP based session affinity.
type ClientIPAffinityConfig struct {
	TimeoutSeconds int32
}
```

What do you think about it? Thanks!